r/vignettes/developers/workflow.Rmd

```diff
@@ -0,0 +1,198 @@
+# Common developer workflow tasks
+
+```{r setup-options, include=FALSE}
+knitr::opts_chunk$set(error = TRUE, eval = FALSE)
+```
+
+The `arrow/r` directory contains a `Makefile` to help with some common tasks from the command line (e.g. `make test`, `make doc`, `make clean`, etc.).
+
+## Loading arrow
+
+You can load the R package via `devtools::load_all()`.
+
+## Rebuilding the documentation
+
+The R documentation uses the [`@examplesIf`](https://roxygen2.r-lib.org/articles/rd.html#functions) tag introduced in `roxygen2` version 7.1.1.9001, which hasn't yet been released on CRAN at the time of writing.  If you are making changes which require updating the documentation, please install the development version of `roxygen2` from GitHub.
+
+```{r, run=FALSE}
+remotes::install_github("r-lib/roxygen2")
```

Comment:
Do we still need this? I know we are waiting on a release, but can't remember if roxygen2 has had one recently or not

Reply:
Yeah, there's a recent update, I'll update this in once I sort the merge conflict.